### PR TITLE
Api documentation update for modal.js

### PR DESCRIPTION
**Pull request by ericnakagawa**
Related to #8203 to update the Modal API reference doc.

**Test plan (required)**

Started up the website and checked:
http://localhost:8079/react-native/docs/modal.html

modal update

**Note, copied from a previous PR**

The code is not Flow-ified so depended on jsdoc formatting to get the method parameter types. There's a current issue with handling optional types via react-docgen which parses components. There's an open PR to look into this: https://github.com/reactjs/react-docgen/pull/89. When that's resolved the `replaceAtIndex` method parameter type that's documented for `cb` needs to be updated to make it optional.

### CODE REVIEW

**Comment by facebook-github-bot-3** [comment]
By analyzing the blame information on this pull request, we identified @jemise111 and @gabelevi to be potential reviewers.

File: Libraries/Modal/Modal.js [hunk 1]
```diff
@@ -23,29 +23,44 @@ const requireNativeComponent = require('requireNativeComponent');
...
- * configureScene property.
  */
 class Modal extends React.Component {
   static propTypes = {
-    animated: deprecatedPropType(
-      PropTypes.bool,
-      'Use the `animationType` prop instead.'
-    ),
+    /**
+     * The `animationType` property controls how the modal animates.
```
Comment (caabernathy) [inline Libraries/Modal/Modal.js]: In general, docs prefer using "prop" instead of "property".

**Comment by caabernathy** [comment]
Could you add a simple Modal example in the intro. Similar to http://facebook.github.io/react-native/releases/next/docs/view.html - just something to quickly show what the code would look like.

File: Libraries/Modal/Modal.js [hunk 2]
```diff
@@ -23,29 +23,94 @@ const requireNativeComponent = require('requireNativeComponent');
...
+ *           <Text>Show Modal</Text>
+ *         </TouchableHighlight>
+ *       </View>
+ *     );
+ *   },
+ * });
+ * 
+ * // App registration and rendering
+ * AppRegistry.registerComponent('AwesomeProject', () => ModalExample);* ```
  */
```
Comment (caabernathy) [inline Libraries/Modal/Modal.js]: Let's make the example ES6 for consistency with the other changes we're making and also drop the AppRegistry part.

**Comment by ericnakagawa** [comment]
@caabernathy I'll keep code in ES6 moving forward.

**Comment by caabernathy** [comment]
Excellent stuff.

**Comment by caabernathy** [comment]
@facebook-github-bot shipit

**Comment by facebook-github-bot-5** [comment]
Thanks for importing. If you are an FB employee go to [Phabricator](https://our.intern.facebook.com/intern/opensource/github/pull_request/291979777858526/int_phab) to review.

**Pull request closed without merging**